test/Removal.t.sol

```diff
@@ -3,12 +3,128 @@
 pragma solidity =0.8.15;
 import "@/test/helpers/market.sol";
 import {InvalidTokenTransfer} from "@/contracts/Errors.sol";
+
 using UInt256ArrayLib for uint256[];
 using AddressArrayLib for address[];
 
 // todo fuzz RemovalIdLib
 // todo test that checks Removal.consign can happen using multi call with mix-match project ids
 
+contract Removal_migrate is UpgradeableMarket {
+  uint256 constant NUMBER_OF_SUPPLIERS = 2;
+  uint256 constant AMOUNT_PER_REMOVAL = 1 ether;
+  uint32 constant NUMBER_OF_REMOVALS_PER_SUPPLIER = 5;
+  uint256 constant EXPECTED_CERTIFICATE_ID = 0;
+
+  function setUp() external {
+    _removal.registerContractAddresses({
+      market: _market,
+      certificate: _certificate
+    });
+    _removal.grantRole({
+      role: _removal.CONSIGNOR_ROLE(),
+      account: _namedAccounts.admin
+    });
+  }
+
+  function test() external {
+    uint256[][] memory amountsForAllSuppliers = new uint256[][](
+      NUMBER_OF_SUPPLIERS
+    );
+    uint256[][] memory idsForAllSuppliers = new uint256[][](
+      NUMBER_OF_SUPPLIERS
+    );
+    address[] memory suppliers = new address[](NUMBER_OF_SUPPLIERS);
+    uint256 numberOfRemovals = NUMBER_OF_SUPPLIERS *
+      NUMBER_OF_REMOVALS_PER_SUPPLIER;
+    uint256[] memory amountsPerSupplier = new uint256[](
+      NUMBER_OF_REMOVALS_PER_SUPPLIER
+    ).fill(AMOUNT_PER_REMOVAL);
+    uint256 certificateAmount = numberOfRemovals * AMOUNT_PER_REMOVAL;
+    for (uint256 i = 0; i < idsForAllSuppliers.length; ++i) {
+      amountsForAllSuppliers[i] = amountsPerSupplier;
+      address supplier = i == 0 ? _namedAccounts.supplier : i == 1
+        ? _namedAccounts.supplier2
+        : _namedAccounts.supplier3;
+      idsForAllSuppliers[i] = _seedRemovals({
+        to: supplier,
+        count: NUMBER_OF_REMOVALS_PER_SUPPLIER,
+        list: false
+      });
+      suppliers[i] = supplier;
+    }
+    vm.recordLogs();
+    vm.prank(_namedAccounts.admin);
+    _removal.migrate({
+      owners: suppliers,
+      ids: idsForAllSuppliers,
+      amounts: amountsForAllSuppliers,
+      certificateRecipient: _namedAccounts.buyer,
+      certificateAmount: certificateAmount
+    });
+    Vm.Log[] memory entries = vm.getRecordedLogs();
+    assertEq(entries.length, 6);
+    assertEq(
+      entries[2].topics[0],
```

Comment:
I think you can use `expectEmit` for that but without refactoring our repo to use interfaces there doesnt seem to be a great way to do that without re-defining the event 

https://book.getfoundry.sh/cheatcodes/expect-emit?highlight=event#examples

Reply:
in either case I think I'd wait for a better test API to come around until preferring one or the other. Keeping this as is so that we have a living example floating around for this approach (we already have examples in other places for expectEmit)